apps/agreement/contracts/executor/IAgreementExecutor.sol

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-License-Identitifer:    GPL-3.0-or-later
+ */
+
+pragma solidity 0.4.24;
+
+import "@aragon/os/contracts/lib/token/ERC20.sol";
+
+
+interface IAgreementExecutor {
+    function pause(uint256 _executableId) external;
+
+    function resume(uint256 _executableId) external;
+
+    function cancel(uint256 _executableId) external;
+
+    function void(uint256 _executableId) external;
+
+    function canChallenge(uint256 _executableId, address _challenger) external view returns (bool);
+
+    function getCollateralRequirements() external view
```

Comment:
Wondering if it'd be worth adding a `bytes4` argument to this, so you could change collateral based on the function you're calling in the "executor".

Reply:
Sure, and we can probably add the `_executableId` as well

Reply:
I've been thinking more about this, not sure which would be the case actually for passing the signature here